### ISSUE TITLE
SIZE intrinsic not compatible with integer(4)

### ISSUE THREAD

**Issue by sscalpone** [flang-compiler/f18#882]
```
module m
contains
  subroutine ai (v, i)
      integer, intent(in) :: i
      integer, intent(in) :: v(:)
  end
end module

subroutine a (v)
    use m
    integer, intent(in) :: v(:)
    call ai(v, size(v))
end
```
```
% flang f.f90
f.f90:12:5: error: actual argument type 'Integer(8)' is not compatible with dummy argument type 'Integer(4)'
      call ai(v, size(v))
      ^^^^^^^^^^^^^^^^^^^
f18: semantic errors in f.f90
```

**Comment by klausler** [comment]
This is a consequence of f18 being in "large array" mode by default.  You can use `SIZE(v,kind=kind(0))` if you want to force `SIZE` or other similar intrinsic functions to not use "large arrays".

**Comment by sscalpone** [comment]
Fixed by c1b200d23891cff4eca1991deec58c1adf65842e
